src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,247 @@
+    shifter_wrapper += f"cd $HOME\n"
+    shifter_wrapper += "shifter --image=$image $command\n"  # Run the command in the Shifter environment
+
+    wrapper_file = os.path.join(job_dir, 'shifter_wrapper.sh')
+    with open(wrapper_file, "w") as f:
+        f.write(shifter_wrapper)
+
+    os.chmod(wrapper_file, 0o777)
+
+    return wrapper_file
+
+
+def _create_genome_id_file(genome_ids, genome_id_file):
+    """
+    Create a tab-separated values (TSV) file with a list of genome IDs.
+    """
+    with open(genome_id_file, 'w') as f:
+        f.write("genome_id\n")
+        for genome_id in genome_ids:
+            f.write(f"{genome_id}\n")
+
+
+def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper_file, job_dir, root_dir):
+    """
+    Create task list file (tasks.txt)
+    """
+    genome_ids = [path for path in os.listdir(source_data_dir) if
+                  os.path.isdir(os.path.join(source_data_dir, path))]
+
+    genome_ids_chunks = [genome_ids[i: i + CHUNK_SIZE] for i in range(0, len(genome_ids), CHUNK_SIZE)]
+
+    task_list = '#!/usr/bin/env bash\n'
+    for idx, genome_ids_chunk in enumerate(genome_ids_chunks):
+
+        task_list += wrapper_file
+
+        genome_id_file = os.path.join(job_dir, f'genome_id_{idx}.tsv')
+        _create_genome_id_file(genome_ids_chunk, genome_id_file)
+
+        if tool == 'checkm2':
+            task_list += f'	--volume={CHECKM2_DB}:/CheckM2_database --volume={root_dir}:/checkm2_root_dir ' \
+                         f'conda run -n checkm2-1.0.0 ' \
+                         f'python collections/src/loaders/genome_collection/compute_genome_attribs.py ' \
+                         f'--tools checkm2 '
+        elif tool == 'gtdb_tk':
+            task_list += f' --volume={GTDBTK_DATA_PATH}:/gtdbtk_reference_data ' \
+                         f'conda run -n gtdbtk-2.1.1 ' \
+                         f'python collections/src/loaders/genome_collection/compute_genome_attribs.py ' \
+                         f'--tools gtdb_tk '
+        else:
+            raise ValueError(f'Unexpected tool {tool}')
+
+        task_list += f'--load_ver {load_ver} --source_data_dir {source_data_dir} ' \
+                     f'--kbase_collection {kbase_collection} --root_dir {root_dir} ' \
+                     f'--threads 128 --program_threads 128 --node_id job_{idx} ' \
+                     f'--debug --source_file_ext genomic.fna.gz --genome_id_file {genome_id_file}\n'
+
+    task_list_file = os.path.join(job_dir, 'tasks.txt')
+    with open(task_list_file, "w") as f:
+        f.write(task_list)
+
+    return task_list_file, len(genome_ids_chunks)
+
+
+def _create_batch_script(job_dir, task_list_file, n_jobs):
+    """
+    Create the batch script (submit_taskfarmer.sl) for job submission
+    """
+    batch_script = '#!/bin/sh\n'
+    batch_script += f'#SBATCH -N {n_jobs + 1} -c 64\n'
+    batch_script += '#SBATCH -q regular\n'
+    batch_script += '#SBATCH --time=2:00:00\n'
```

Comment:
This is enough time for 1k genomes thru gtdb_tk?

Reply:
I changed the job duration to 4 hours, but I believe 2 hours would suffice. Assigning shorter time intervals to jobs can decrease the waiting period for jobs. Therefore, I allocate the minimum time possible for a job.

Reply:
If you think 2 hours is fine we should go with two hours. I'd just seen Shane use that randomly and so I was wondering if it was just copied from there or if we were sure 2 was ok

Reply:
I will leave it for 4 hours just for now. I think we might want to make it adjustable depending on tools in the future.

Reply:
:+1: